### ISSUE TITLE
Assertion error sometimes

### ISSUE THREAD

**Issue by pgimeno** [shlomif/PySolFC#12]
When playing PileOn or Small PileOn, (Select > French Games > One-Deck Games > Small PileOn), in rare occasions an assertion error can be triggered.

The assertion says `assert c.face_up` and as of this writing (commit 29a9a61) is in line 520 of game.py, visible here:

https://github.com/shlomif/PySolFC/blob/29a9a61eef93e2376ead5f3d5001f254955c4031/pysollib/game.py#L514

The reason is that occasionally (very rarely, but it has happened to me twice in a session) all four cards in a stack are equal, causing the stack to face down.

It is easier to reproduce with a reduced version, e.g. adding this to `pysollib/game/pileon.py`:

``` python
class SmallerPileOn(PileOn):
    TWIDTH = 3
    NSTACKS = 5
    PLAYCARDS = 4

registerGame(GameInfo(10001, SmallerPileOn, "Smaller PileOn",
                      GI.GT_1DECK_TYPE | GI.GT_OPEN | GI.GT_ORIGINAL, 1, 0, GI.SL_MOSTLY_SKILL,
                      ranks=tuple([0]+range(16-SmallerPileOn.NSTACKS, 13)),
                      #(0, 10, 11, 12),
                      rules_filename = "pileon.html"))
```

and then selecting _Smaller PileOn_ and pressing 'N' repeatedly.

Possible solutions that I can think of are to avoid generating such a deal (e.g. by redealing in case this situation is detected) and to remove the assertion.

**Comment by shlomif** [comment]
Thanks for the report! I'll try to investigate soon.

**Comment by shlomif** [comment]
Fixed in commit cc09f7a5d603b42efaef16038a43beec4451a299

**Comment by shlomif** [comment]
Closing as fixed, thanks!
